src/client/components/atoms/burgerButton/index.tsx

```diff
@@ -0,0 +1,52 @@
+import './styles.scss';
+import { useContext, useState } from 'react';
+import { Link } from 'react-router-dom';
+import StateContext from '../../../utils/stateContext';
+import ExpandMenu from '../collapse/expandMenu';
+
+const BurgerButton = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { user } = useContext(StateContext);
+
+  const handleToggleMenu = () => {
+    setIsOpen(!isOpen);
+  };
+  return (
+    <>
+      <button className={`burger-button ${isOpen ? 'open' : ''}`} onClick={handleToggleMenu}>
```

Comment:
1. Here we are not properly applying BEM pattern, this is a perfect place for a modifier
2. We use `classnames` library for dynamic classNames

suggestion:

`<button className={classnames("burger-menu__button", { "burger-menu__button--open": isOpen })} .....>`

- `<button className={`burger-button ${isOpen ? 'open' : ''}`} `